qiime/plugin/testing.py

```diff
@@ -84,3 +86,17 @@ def assertSemanticTypeRegisteredToFormat(self, semantic_type, exp_format):
             obs_format, exp_format,
             "Expected semantic type %r to be registered to format %r, not %r."
             % (semantic_type, exp_format, obs_format))
+
+    def transformer_helper(self, source, target, filenames):
+        transformer = self.get_transformer(source, target)
+
+        for filename in filenames:
+            filepath = self.get_data_path(filename)
+            shutil.copy(filepath, self.temp_dir.name)
+        input = source(self.temp_dir.name, mode='r')
+
+        obs = transformer(input)
+
+        self.assertIsInstance(obs, target)
```

Comment:
if `target` is a (`pathlib.Path`, `str`, or a subclass of `DirectoryFormat`), check that `obs` is an instance of `pathlib.Path`, `str`, or a subclass of `DirectoryFormat`. If not, just `assertIsInstance(obs, target)`.